tests/main/snap-discard-ns/task.yaml

```diff
@@ -7,17 +7,30 @@ details: |
 prepare: |
     . $TESTSLIB/snaps.sh
     install_local test-snapd-tools
+environment:
+    # TODO: vary by distro later
+    LIBEXECDIR: /usr/lib/snapd/
 execute: |
-    echo "We can discard the namespace before a snap app runs without errors"
-    /usr/lib/snapd/snap-discard-ns test-snapd-tools
+    echo "We can try to discard a namespace before snap runs"
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
 
-    test-snapd-tools.success
+    echo "We can try to discard a namespace before the .mnt file exits"
+    mkdir -p /run/snapd/ns/
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
+
+    echo "We can try to discard a namespace before the .mnt file is mounted"
+    touch /run/snapd/ns/test-snapd-tools.mnt
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
 
-    echo "We can discard the namespace after a snap app runs without errors"
-    /usr/lib/snapd/snap-discard-ns test-snapd-tools
+    echo "We can discard the namespace after a snap runs"
+    test-snapd-tools.success
+    # The last hex is the same as nsfs but older stat on ubuntu 14.04 doesn't know
+    stat -f -c %T /run/snapd/ns/test-snapd-tools.mnt | MATCH 'proc|nsfs|0x6e736673'
```

Comment:
Also, I understand why 'nsfs' and '0x6e736673', but why 'proc'? If that is intentional, please comment on why.